tests/test_utils.py

```diff
@@ -68,3 +71,145 @@ def test_validate_against_schema(instance):
+
+
+@pytest.mark.parametrize(
+    "instance,no_errs,bad_kw",
+    [
+        ("bad_metadata/metadata-missing-name.yaml", 1, []),
+        ("bad_metadata/metadata-missing-name-missing-uuid.yaml", 2, []),
+        (
+            "bad_metadata/metadata-bad-name-missing-uuid.yaml",
+            2,
+            [
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-name.yaml",
+            1,
+            [
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-name2.yaml",
+            1,
+            [
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-name2-missing-uuid.yaml",
+            2,
+            [
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-descript.yaml",
+            1,
+            [
+                "description",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-descript2.yaml",
+            1,
+            [
+                "description",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-longdescript.yaml",
+            1,
+            [
+                "long_description",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-longdescript2.yaml",
+            1,
+            [
+                "long_description",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-descript-bad-name.yaml",
+            2,
+            [
+                "description",
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-descript-bad-name2.yaml",
+            2,
+            [
+                "description",
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-descript2-bad-name2.yaml",
+            2,
+            [
+                "description",
+                "name",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-all-bad.yaml",
+            4,
+            [
+                "description",
+                "name",
+                "long_description",
+                "experiment_uuid",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-created.yaml",
+            1,
+            [
+                "created",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-url.yaml",
+            1,
+            [
+                "url",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-contact.yaml",
+            1,
+            [
+                "contact",
+            ],
+        ),
+        (
+            "bad_metadata/metadata-bad-email.yaml",
+            1,
+            [
+                "email",
+            ],
+        ),
+    ],
+)
+def test_bad_metadata_validation_raises(test_data, instance, no_errs, bad_kw):
```

Comment:
I think it would be better to rewrite this as 
```python
    fpath = str(test_data / Path(instance))

    with pytest.raises(jsonschema.ValidationError) as excinfo:
        _ = load_metadata_yaml(fpath, EXP_JSONSCHEMA)

    err_msg = excinfo.value.message

    assert (
        f"{no_errs:02d}" in err_msg and f"{no_errs+1:02d}" not in err_msg
    ), "Can't see the right number of errors!"
    for kw in bad_kw:
        assert f"| {kw}" in err_msg, f"Can't see expected specific error for {kw}"
```
& let pytest handle the exception

Reply:
Noticed this whilst investigating local machine test failures - it was just an environment conflict.

Other than this, looks ready to merge.

Reply:
You mean let pytest just spit out the unexpected exception if it doesn't send back `jsonschema.ValidationError` for whatever reason?

Reply:
More or less - it looks to me like lines 212-215 you're manually handling all exceptions other than jsonschema.Validation error (sorry, highlighted the wrong lines in the original comment) in order to raise that we got an unexpected exception? 

It feels cleaner to me to let pytest handle that for us.

Reply:
OK that's done.